edgepi_rpc_client/services/pwm/client_pwm_service.py

```diff
@@ -0,0 +1,90 @@
+"""Client for PWM service"""
+import logging
+from edgepirpc.protos import pwm_pb2 as pwm_pb
+from edgepi_rpc_client.client_rpc_channel import ClientRpcChannel
+from edgepi_rpc_client.util.helpers import (
+    filter_arg_values, create_config_request_from_args, get_server_response
+)
+from edgepi_rpc_client.services.pwm.pwm_pb_enums import PWMPins, Polarity
+
+_logger = logging.getLogger(__name__)
+
+# pylint: disable=no-member
+class ClientPWMService():
+    """Client Methods for PWM Service"""
+    def __init__(self, transport):
+        self.client_rpc_channel = ClientRpcChannel(transport)
+        self.service_stub = pwm_pb.PWMService_Stub(self.client_rpc_channel)
+        self.rpc_controller = None
+
+    # pylint: disable=unused-argument
+    def set_config(self, pwm_num: PWMPins,
+                   frequency: float = None,
+                   duty_cycle: float = None,
+                   polarity: Polarity = None):
+        """set_config method for SDK PWM module"""
+        config_args_dict = filter_arg_values(locals(), 'self', None)
+        config_msg = pwm_pb.Config()
+        arg_msg = pwm_pb.Config().ConfArg()
+
+        request = create_config_request_from_args(config_msg, arg_msg, config_args_dict)
+        rpc_response = self.service_stub.set_config(self.rpc_controller,request)
+        response = get_server_response(rpc_response, pwm_pb.SuccessMsg)
+        return response.content
+
+    def enable(self, pwm_num: PWMPins):
+        """enable method for SDK PWM module"""
+        request = pwm_pb.PWM(pwm_num = pwm_num.value)
+        rpc_response = self.service_stub.enable(self.rpc_controller, request)
+        response = get_server_response(rpc_response, pwm_pb.SuccessMsg)
+        return response.content
+
+    def disable(self, pwm_num: PWMPins):
+        """disable method for SDK PWM module"""
+        request = pwm_pb.PWM(pwm_num = pwm_num.value)
+        rpc_response = self.service_stub.disable(self.rpc_controller, request)
+        response = get_server_response(rpc_response, pwm_pb.SuccessMsg)
+        return response.content

```

Comment:
All the methods here look pretty similar for PWM, maybe a helper function like this could simplify the code a bit?

An idea:
```
def call_pwm_client_method(self, pwm_num, req_method, server_resp_class):
    request = pwm_pb.PWM(pwm_num = pwm_num.value)
    rpc_response = req_method(self.rpc_controller, request)
    return get_server_response(rpc_response, server_resp_class)
```

Then your individual methods could look like:
```
def disable(self, pwm_num: PWMPins):
    """disable method for SDK PWM module"""
    response = self.call_pwm_client_method(
        pwm_num, self.service_stub.disable, pwm_pb.SuccessMsg
    )
    return response.content
```

Reply:
Refactored. Thanks!